app/electron/main.ts

```diff
@@ -98,6 +99,270 @@ const buildManifest = fs.existsSync(manifestFile) ? require(manifestFile) : {};
+      action: 'INSTALL' | 'UNINSTALL' | 'UPDATE' | 'LIST' | 'CANCEL';
+      progress: ProgressResp;
+      controller?: AbortController;
+      percentage?: number;
+    };
+  } = {};
+
+  constructor() {
+    this.cache = {};
+  }
+
+  /**
+   * Converts the progress response to a percentage.
+   *
+   * @param {ProgressResp} progress - The progress response object.
+   * @returns {number} The progress as a percentage.
+   */
+  private convertProgressToPercentage(progress: ProgressResp): number {
+    switch (progress.message) {
+      case 'Fetching Plugin Metadata':
+        return 20;
+      case 'Plugin Metadata Fetched':
+        return 30;
+      case 'Downloading Plugin':
+        return 50;
+      case 'Plugin Downloaded':
+        return 100;
+      default:
+        return 0;
+    }
+  }
+
+  /**
+   * Sets up event handlers for plugin-manager.
+   *
+   * @method
+   * @name setupEventHandlers
+   */
+  setupEventHandlers() {
+    ipcMain.on('plugin-manager', (event, data) => {
+      const eventData = JSON.parse(data) as Action;
+      const { identifier, action } = eventData;
+
+      const updateCache = (progress: ProgressResp) => {
+        const percentage = this.convertProgressToPercentage(progress);
+        this.cache[identifier].progress = progress;
+        this.cache[identifier].percentage = percentage;
+      };
+
+      switch (action) {
+        case 'INSTALL':
+          this.handleInstall(eventData, updateCache);
+          break;
+        case 'UPDATE':
+          this.handleUpdate(eventData, updateCache);
+          break;
+        case 'UNINSTALL':
+          this.handleUninstall(eventData, updateCache);
+          break;
+        case 'LIST':
+          this.handleList(event, eventData);
+          break;
+        case 'CANCEL':
+          this.handleCancel(event, identifier);
+          break;
+        case 'GET':
+          this.handleGet(event, identifier);
+          break;
+        default:
+          console.error(`Unknown action: ${action}`);
+      }
+    });
+  }
+
+  /**
+   * Handles the installation process.
+   *
+   * @method
+   * @name handleInstall
+   * @private
+   */
+  private handleInstall(eventData: Action, updateCache: (progress: ProgressResp) => void) {
+    const { identifier, URL, destinationFolder, headlampVersion } = eventData;
+    if (!URL) {
+      this.cache[identifier] = {
+        action: 'INSTALL',
+        progress: { type: 'error', message: 'URL is required' },
+      };
+      return;
+    }
+
+    const controller = new AbortController();
+    this.cache[identifier] = {
+      action: 'INSTALL',
+      progress: { type: 'info', message: 'installing plugin' },
+      percentage: 10,
+      controller,
+    };
+
+    PluginManager.install(
+      URL,
+      destinationFolder,
+      headlampVersion,
+      progress => {
+        updateCache(progress);
+      },
+      controller.signal
+    );
+  }
+
+  /**
+   * Handles the update process.
+   *
+   * @method
```

Comment:
I wonder if method, name and private are needed? What do they add compared to if they are missing?

Reply:
These tags are part of jsdoc and provide enhanced support in editors.
![image](https://github.com/headlamp-k8s/headlamp/assets/22731013/74d4d057-95af-43cb-b91f-7b162fc34fce)